src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs

```diff
@@ -873,13 +874,60 @@ public unsafe IMonitor Monitor
 
                 var monitor = _glfw.GetWindowMonitor(_windowPtr);
                 if (monitor == null)
-                    monitor = _glfw.GetPrimaryMonitor();
+                {
+                    if (_windowState != WindowState.Fullscreen)
+                    {
+                        // Determine which monitor this window is on. [6 marks]
+                        var monitors = new GlfwMonitorEnumerable();
+                        foreach (var m in monitors)
+                        {
+                            if (m.Bounds.Contains(Position))

```

Comment:
This wouldn't return true if the window corner is offscreen. Isn't it better to use the window's center? Even better, when the position isn't on any particular monitor, you test which is closest.

Reply:
Yeah you're right, I'll change it to use the centre and try and add in an extra test.

Reply:
Or the most overlap of window and monitor. Like shown here: https://stackoverflow.com/questions/21421074/how-to-create-a-full-screen-window-on-the-current-monitor-with-glfw

Reply:
I've made it use the centre and attempted to come up with a check test, but I didn't like it as it felt overengineered plus I'm not that well-versed with the maths, so I don't feel too confident writing such a solution. I've committed it but only so you could see my potential solution - I immediately reverted it.